packages/derived-wallet-solana/src/signAptosTransaction.ts

```diff
@@ -46,31 +45,58 @@ export async function signAptosTransactionWithSolana(input: SignAptosTransaction
     domain
   });
 
-  const response = await wrapSolanaUserResponse(solanaWallet.signIn!(siwsInput));
-  return mapUserResponse(response, (output): AccountAuthenticator => {
-    if (output.signatureType && output.signatureType !== 'ed25519') {
-      throw new Error('Unsupported signature type');
-    }
+  // Prioritize SIWS if available
+  if(solanaWallet.signIn){
+
+    const response = await wrapSolanaUserResponse(solanaWallet.signIn!(siwsInput));
+    return mapUserResponse(response, (output): AccountAuthenticator => {
+      if (output.signatureType && output.signatureType !== 'ed25519') {
+        throw new Error('Unsupported signature type');
+      }
+  
+      // The wallet might change some of the fields in the SIWS input, so we
+      // might need to include the finalized input in the signature.
+      // For now, we can assume the input is unchanged.
+      const signature = new Ed25519Signature(output.signature);
+  
+      return createAccountAuthenticatorForSolanaTransaction(signature, solanaPublicKey, domain, authenticationFunction, signingMessageDigest);
+    });
+  }else if(solanaWallet.signMessage){
+    // Fallback to signMessage if SIWS is not available
+    const response = await wrapSolanaUserResponse(solanaWallet.signMessage(createSignInMessage(siwsInput)));
+    return mapUserResponse(response, (output): AccountAuthenticator => {
+      
+      // Solana signMessage standard always returns a Ed25519 signature type
+      const signature = new Ed25519Signature(output);
 
-    // The wallet might change some of the fields in the SIWS input, so we
-    // might need to include the finalized input in the signature.
-    // For now, we can assume the input is unchanged.
-    const signature = new Ed25519Signature(output.signature);
+      return createAccountAuthenticatorForSolanaTransaction(signature, solanaPublicKey, domain, authenticationFunction, signingMessageDigest);
+    });
+  }else{
+    throw new Error(`${solanaWallet.name} does not support SIWS or signMessage`);
+  }
+}
 
-    // Serialize the signature with the signature type as the first byte.
-    const serializer = new Serializer();
-    serializer.serializeU8(SIGNATURE_TYPE);
-    serializer.serializeBytes(signature.toUint8Array());
-    const abstractSignature = serializer.toUint8Array();
+// A helper function to create an AccountAuthenticator from a Solana signature
+function createAccountAuthenticatorForSolanaTransaction(
+  signature: Ed25519Signature, 
+  solanaPublicKey: SolanaPublicKey, 
+  domain: string, 
+  authenticationFunction: string, 
+  signingMessageDigest: Uint8Array
+) : AccountAuthenticator {
+  // Serialize the signature with the signature type as the first byte.
+  const serializer = new Serializer();
+  serializer.serializeU8(SIGNATURE_TYPE);
+  serializer.serializeBytes(signature.toUint8Array());
+  const abstractSignature = serializer.toUint8Array();
```

Comment:
should probably define a class for this in the future :)

```typescript
export class SolanaAsbtractSignature {
  static readonly DEFAULT_TYPE = 0;
  constructor(data: Uint8Array) { ... }
  serialize(...) { }
}
```

unrelated to this pr